3main

```diff
@@ -1040,80 +1042,57 @@ then
 
   TI="3"
 
-  if [[ $___1050_or_1050ti == YES ]]
+  if [ $___1050_or_1050ti == "YES" ]
   then
     TI="2"
-    if [[ $NORMAL == YES ]]
-    then
-      TI="2 3"
-    fi
+  if  
+  if [ $NORMAL == "YES" ]
+  then
+    TI="2 3"
   fi
 
   if [[ $SLOW_USB_KEY_MODE == YES ]]
   then
     sleep 6
   fi
 
-  if [[ $P10X_100_FULL_HEADLESS_MODE == NO ]]
+  if [[ $MANUAL_FAN == YES && $MINER_TEMP_CONTROL == NO ]]
   then
+    gpu=0
+    while (( gpu < GPUS ))
+    do
+      NVD_SETTINGS="${NVD_SETTINGS} -a [gpu:$gpu]/GPUFanControlState=1 -a [fan:$gpu]/GPUTargetFanSpeed=${FAN_SPEED}"
+      gpu=$((gpu+1))
+    done
+  fi
 
-    if [[ $MANUAL_FAN == YES && $MINER_TEMP_CONTROL == NO ]]
-    then
-      gpu=0
-      while (( gpu < GPUS ))
-      do
-        NVD_SETTINGS="${NVD_SETTINGS} -a [gpu:$gpu]/GPUFanControlState=1 -a [fan:$gpu]/GPUTargetFanSpeed=${FAN_SPEED}"
-        gpu=$((gpu+1))
-      done
-    fi
+  if [[ $MANUAL_FAN == NO && $MINER_TEMP_CONTROL == NO ]]
+  then
+    gpu=0
+    while (( gpu < GPUS ))
+    do
+      NVD_SETTINGS="${NVD_SETTINGS} -a [gpu:$gpu]/GPUFanControlState=0"
+      gpu=$((gpu+1))
+    done
+  fi
 
-    if [[ $MANUAL_FAN == NO && $MINER_TEMP_CONTROL == NO ]]
-    then
-      gpu=0
-      while (( gpu < GPUS ))
+  if [[ $HEADLESS_MODE == NO ]]
```

Comment:
So We should remove the statements for headless mode and run the part from headless == no ?

Reply:
Does the TI thingy works with p10x too?

Reply:
full_headless = no does not mean no p10x cards are installed, and if it worked fine with p10x cards till now was because of the TI workaround

Up to now full_headless is implicitly considering all cards as p10x, but in theory one could be able (for misterious reasons) to activate full_headless with other cards having video output and the current code would work in such a scenario if all cards are either p10x or TI cards

If I understood correctly, should be fine to simply remove line originally at 1117, now at 1096 together with its surrounding if block, and also put code block from 1080 to 1091 outside of its if 
statement